astropy/nddata/utils.py

```diff
@@ -25,6 +28,16 @@
     "Cutout2D",
 ]
 
+if TYPE_CHECKING:
+    from numbers import Number
+    from typing import Literal, TypeAlias
+
+    from typing_extensions import TypeVar
+
+    T = TypeVar("T")
+    SingleOrTuple: TypeAlias = T | tuple[T, ...]
```

Comment:
This [`TypeAlias`](https://docs.python.org/3/library/typing.html#typing.TypeAlias) is causing problems for Sphinx. Luckily `int | tuple[int, ...]` or `float | tuple[float, ...]` are so simple that we don't really need an alias with a [`TypeVar`](https://docs.python.org/3/library/typing.html#typing.TypeVar) for them.